tests/deploy/test-communicator/main.go

```diff
@@ -0,0 +1,566 @@
+	Timestamp string `json:"timestamp"`
+}
+
+type DataResponse struct {
+	Message   string                 `json:"message"`
+	Service   string                 `json:"service"`
+	Timestamp string                 `json:"timestamp"`
+	Data      map[string]interface{} `json:"data"`
+}
+
+type UserResponse struct {
+	ID      string `json:"id"`
+	Name    string `json:"name"`
+	Service string `json:"service"`
+	Created string `json:"created"`
+}
+
+func NewApp() *App {
+	config := Config{
+		Port:        getEnvAsInt("PORT", 8080),
+		ServiceName: getEnv("SERVICE_NAME", "test-communicator"),
+		TargetURL:   getEnv("TARGET_URL", ""),
+		Protocol:    getEnv("PROTOCOL", "http"),
+		TargetHost:  getEnv("TARGET_HOST", ""),
+		TargetPort:  getEnvAsInt("TARGET_PORT", 8080),
+	}
+
+	app := &App{
+		config: config,
+		router: mux.NewRouter(),
+		stopCh: make(chan struct{}),
+	}
+
+	// Initialize Prometheus metrics
+	app.requests = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "requests_total",
+		Help: "Total number of requests",
+	})
+	prometheus.MustRegister(app.requests)
+
+	// Setup HTTP routes if HTTP protocol is enabled
+	if config.Protocol == "http" || config.Protocol == "all" {
+		app.setupHTTPRoutes()
+	}
+
+	return app
+}
+
+func (a *App) setupHTTPRoutes() {
+	// Health endpoint
+	a.router.HandleFunc("/health", a.healthHandler).Methods("GET")
+
+	// API endpoints
+	a.router.HandleFunc("/api/data", a.dataHandler).Methods("GET")
+	a.router.HandleFunc("/api/users/{id}", a.userHandler).Methods("GET")
+	a.router.HandleFunc("/api/call-target", a.callTargetHandler).Methods("GET")
+
+	// Metrics endpoint
+	a.router.Handle("/metrics", promhttp.Handler())
+
+	// Root endpoint
+	a.router.HandleFunc("/", a.rootHandler).Methods("GET")
+
+	// Add middleware for request counting
+	a.router.Use(a.requestCounterMiddleware)
+}
+
+func (a *App) requestCounterMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		a.requests.Inc()
+		next.ServeHTTP(w, r)
+	})
+}
+
+func (a *App) healthHandler(w http.ResponseWriter, r *http.Request) {
+	response := HealthResponse{
+		Status:    "healthy",
+		Service:   a.config.ServiceName,
+		Timestamp: time.Now().UTC().Format(time.RFC3339),
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(response)
+}
+
+func (a *App) dataHandler(w http.ResponseWriter, r *http.Request) {
+	response := DataResponse{
+		Message:   "Data retrieved successfully",
+		Service:   a.config.ServiceName,
+		Timestamp: time.Now().UTC().Format(time.RFC3339),
+		Data: map[string]interface{}{
+			"items":  []string{"item1", "item2", "item3"},
+			"count":  3,
+			"active": true,
+		},
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(response)
+}
+
+func (a *App) userHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	userID := vars["id"]
+
+	response := UserResponse{
```

Comment:
Why is this message so complex? And do we even use it for anything?

Reply:
Removing `Name` which is unused